src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1000,11 +1004,17 @@ async Task WriteExecute(NpgsqlConnector connector, bool async, bool flush, Cance
 
                 await connector.WriteExecute(0, async, cancellationToken);
 
+                if (batchCommand.AppendErrorBarrier ?? EnableErrorBarriers)
+                    await connector.WriteSync(async, cancellationToken);
+
                 if (pStatement != null)
                     pStatement.LastUsed = DateTime.UtcNow;
             }
 
-            await connector.WriteSync(async, cancellationToken);
+            if (batchCommand is null || !(batchCommand.AppendErrorBarrier ?? EnableErrorBarriers))
```

Comment:
In which case we can attempt to send 0 queries (so `batchCommand is null` is `true`)?

Reply:
This is the degenerate case where the user executes a batch with no commands, or a command with an empty CommandText. It's a bit silly, I know - I did experiment with not sending anything to PG in this case, but it's a little tricky with multiplexing, which waits for ExecutionCompletion etc., so we would have to add conditions there instead. It didn't seem worth trying to optimize...